client/lib/site/test/test.js

```diff
@@ -1,101 +1,67 @@
+
+import Site from '..';
+
+describe( 'Calypso Site', () => {
+	describe( 'setting attributes', () => {
+		let site;
+		beforeEach( () => {
+			site = Site( {
 				ID: 1234,
 				name: 'Hello',
 				description: 'Hunting bugs is fun.'
 			} );
+		} );
 
+		it( 'attribute changed', () => {
 			site.set( { name: 'Goodbye' } );
-
-			assert( site.name === 'Goodbye' );
+			expect( site.name ).to.equal( 'Goodbye' );
 		} );
 
-		it( 'change event fires on attribute change', function() {
-			var called = false,
-				site;
-
-			site = Site( {
-				ID: 1234,
-				name: 'Hello',
-				description: 'Hunting bugs is fun.'
-			} );
-
-			site.once( 'change', function() {
-				called = true;
-			} );
-
+		it( 'change event fires on attribute change', () => {
+			let changeCallback = sinon.spy();
+			site.on( 'change', changeCallback );
 			site.set( { name: 'Goodbye' } );
-
-			assert( called === true );
+			expect( changeCallback.called ).to.be.true;
 		} );
 
-		it( "change doesn't fire when setting attribute to same value", function() {
-			var called = false,
-				site;
-
-			site = Site( {
-				ID: 1234,
-				name: 'Hello',
-				description: 'Hunting bugs is fun.'
-			} );
-
-			site.once( 'change', function() {
-				called = true;
-			} );
-
+		it( "change doesn't fire when setting attribute to same value", () => {
+			let changeCallback = sinon.spy();
+			site.once( 'change', changeCallback );
 			site.set( { name: 'Hello' } );
-
-			assert( called === false );
+			expect( changeCallback.called ).to.be.false;
 		} );
 
-		it( "change doesn't fire when attribute is set to an object with identical data", function() {
-			var called = false,
-				site;
-
-			site = Site( {
-				ID: 1234,
-				obj: {
-					name: 'Hello',
-					description: 'Hunting bugs is fun.'
-				}
-			} );
+		it( "change doesn't fire when attribute is set to an object with identical data", () => {
+			let changeCallback = sinon.spy();
+			let nestedObj = {
+				name: 'Hello, again.',
+				description: 'Still hunting bugs.'
+			}
+			site.set( { nestedObj } );
 
-			site.once( 'change', function() {
-				called = true;
-			} );
+			site.once( 'change', changeCallback );
 
-			site.set( { obj: {
-					name: 'Hello',
-					description: 'Hunting bugs is fun.'
+			site.set( {
+				nestedObj: {
+					name: 'Hello, again.',
+					description: 'Still hunting bugs.'
 				}
 			} );
-
-			assert( called === false );
+			expect( changeCallback.called ).to.be.false;
 		} );
 
-		it( "change doesn't fire when attribute is set to an array with identical data", function() {
-			var called = false,
-				site;
-
-			site = Site( {
-				ID: 1234,
-				arr: [ 1, 2, 3 ]
-			} );
-
-			site.once( 'change', function() {
-				called = true;
-			} );
-
+		it( "change doesn't fire when attribute is set to an array with identical data", () => {
+			let changeCallback = sinon.spy();
 			site.set( { arr: [ 1, 2, 3 ] } );
 
-			assert( called === false );
+			site.once( 'change', changeCallback );
```

Comment:
Looking at this test I think tests were easier to read when `Site` instance were created in each test.